src/main.cpp

```diff
@@ -28,26 +33,64 @@ int main(void)
         glfwTerminate();
         exit(EXIT_FAILURE);
     }
-    glfwSetKeyCallback(window, key_callback);
+   
     glfwMakeContextCurrent(window);
 	gladLoadGLLoader((GLADloadproc)glfwGetProcAddress);
-    glfwSwapInterval(1);
-    // NOTE: OpenGL error checks have been omitted for brevity
```

Comment:
The OpenGL error checks should probably be "un-omitted" and this is a good opportunity for it :)